File: SDKs/googleapis.com/appengine/v1beta/go/pkg/models/shared/healthcheck.go
package shared

// HealthCheck
// Health checking configuration for VM instances. Unhealthy instances are killed and replaced with new instances. Only applicable for instances in App Engine flexible environment.
type HealthCheck struct {
	CheckInterval      *string `json:"checkInterval,omitempty"`
	DisableHealthCheck *bool   `json:"disableHealthCheck,omitempty"`
	HealthyThreshold   *int64  `json:"healthyThreshold,omitempty"`
	Host               *string `json:"host,omitempty"`
	RestartThreshold   *int64  `json:"restartThreshold,omitempty"`
	Timeout            *string `json:"timeout,omitempty"`
	UnhealthyThreshold *int64  `json:"unhealthyThreshold,omitempty"`
}

File: SDKs/googleapis.com/appengine/v1beta/typescript/src/sdk/models/shared/healthcheck.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// HealthCheck
/** 
 * Health checking configuration for VM instances. Unhealthy instances are killed and replaced with new instances. Only applicable for instances in App Engine flexible environment.
**/
export class HealthCheck extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=checkInterval" })
  checkInterval?: string;

  @SpeakeasyMetadata({ data: "json, name=disableHealthCheck" })
  disableHealthCheck?: boolean;

  @SpeakeasyMetadata({ data: "json, name=healthyThreshold" })
  healthyThreshold?: number;

  @SpeakeasyMetadata({ data: "json, name=host" })
  host?: string;

  @SpeakeasyMetadata({ data: "json, name=restartThreshold" })
  restartThreshold?: number;

  @SpeakeasyMetadata({ data: "json, name=timeout" })
  timeout?: string;

  @SpeakeasyMetadata({ data: "json, name=unhealthyThreshold" })
  unhealthyThreshold?: number;
}
